### PR TITLE
[11.0][ADD] hotel_reservation: improving reservations and reports

### PR DESCRIPTION
**Pull request by mclaeysb**
This pr will improve reservations and improve reservation reports

- [x] `note` field on reservations.
- [x] `note` and `open` fields on reservation lines. These will be put on a per-room report to inform cleaning staff
- [x] add mail thread to reservation form view
- [x] improve reservation reports

### CODE REVIEW

**Comment by mclaeysb** [comment]
cc @robinkeunen

**Comment by mclaeysb** [comment]
One commit expected to make checking report landscape

File: hotel_reservation/views/hotel_reservation.xml [hunk 1]
```diff
@@ -60,6 +60,10 @@
                             <field name="checkout"/>
                             <field name="adults" required="1"/>
                             <field name="children"/>
+                            <field name="open"/>
+                        </group>
+                        <group>
+                            <field name="note"/>
```
Comment (robinkeunen) [inline hotel_reservation/views/hotel_reservation.xml]: I'd put this note in the "note group" and add a name attribute to that group.

I'll put my housekeeping note in that group as well.

Comment (mclaeysb) [reply hotel_reservation/views/hotel_reservation.xml]: Done! Your housekeeping note is welcome to join!

**Pull request merged**